src/components/KFileUpload/KFileUpload.vue

```diff
@@ -155,73 +102,82 @@ const props = defineProps({
     type: Number,
     default: null,
   },
-  /**
-  * Set icon size
-  */
-  iconSize: {
-    type: String,
-    default: KUI_ICON_SIZE_50,
-  },
-  icon: {
-    type: String,
-    default: 'image',
-  },
-  /**
-  * Set icon color
-  */
-  iconColor: {
-    type: String,
-    default: undefined,
-  },
-  hasError: {
+  error: {
     type: Boolean,
     default: false,
   },
   errorMessage: {
     type: String,
-    default: 'Please check file size.',
+    default: '',
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
   },
 })
 
+const attrs = useAttrs()
+const slots = useSlots()
+
 const emit = defineEmits<{
   (e: 'file-added', val: File[]): void
   (e: 'file-removed'): void
   (e: 'error', val: File[]): void
 }>()
 
-const customInputId = computed((): string => props.testMode ? 'test-file-upload-id-1234' : uuidv1())
+const { stripRequiredLabel } = useUtilities()
+
+const inputId = computed((): string => attrs.id ? String(attrs.id) : uuidv4())
+const hasLabelTooltip = computed((): boolean => !!(props.labelAttributes?.info || slots['label-tooltip']))
+const strippedLabel = computed((): string => stripRequiredLabel(props.label, isRequired.value))
+const isRequired = computed((): boolean => attrs?.required !== undefined && String(attrs?.required) !== 'false')
+
+const hasFileSizeError = ref<boolean>(false)
+const fileSizeErrorMessage = computed((): string => {
+  if (hasFileSizeError.value) {
+    return `File size must be less than ${maximumFileSize.value} bytes.`
+  }
+
+  return ''
+})
 const maximumFileSize = computed((): Number => {
   if (props.maxFileSize || props.maxFileSize === 0) {
     return props.maxFileSize
   }
-  return props.type === 'file' ? 5250000 : 1000000
+
+  return 5250000
 })
 
-const hasUploadError = ref(false)
+const hasUploadError = ref<boolean>(false)
 
 // This holds the FileList
 const fileInput = ref<File[]>([])
 // To clear the input value after reset
 const fileInputKey = ref(0)
 // File fakepath
-const fileValue = ref('')
+const fileValue = ref<string>('')
 // Array to store the previously selected FileList when user clicks reopen the file uploader and clicks on Cancel
 const fileClone = ref<File[]>([])
 
 const onFileChange = (evt: any): void => {
   fileInput.value = evt.target?.files
-  fileValue.value = fileInput?.value[0]?.name
+  fileValue.value = String(fileInput?.value[0]?.name)
 
   const fileSize = fileInput?.value[0]?.size
 
   hasUploadError.value = Number(fileSize) as Number > maximumFileSize.value
 
   if (hasUploadError.value) {
     fileInputKey.value++
+
+    if (Number(fileSize) as Number > maximumFileSize.value) {
```

Comment:
```typescript
    if (Number(fileSize) as Number > maximumFileSize.value) {
```

Do you need `as Number` if you're forcing a number before?

Reply:
Fixed